purchases-capacitor-ui/android/src/main/java/com/revenuecat/purchases/capacitor/ui/RevenueCatUIPlugin.kt

```diff
@@ -0,0 +1,149 @@
+package com.revenuecat.purchases.capacitor.ui
+
+import android.content.Intent
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.fragment.app.FragmentActivity
+import com.getcapacitor.JSObject
+import com.getcapacitor.Plugin
+import com.getcapacitor.PluginCall
+import com.getcapacitor.PluginMethod
+import com.getcapacitor.annotation.CapacitorPlugin
+import com.revenuecat.purchases.hybridcommon.ui.PaywallResultListener
+import com.revenuecat.purchases.hybridcommon.ui.PaywallSource
+import com.revenuecat.purchases.hybridcommon.ui.PresentPaywallOptions
+import com.revenuecat.purchases.hybridcommon.ui.presentPaywallFromFragment
+import com.revenuecat.purchases.ui.revenuecatui.customercenter.ShowCustomerCenter
+
+@CapacitorPlugin(name = "RevenueCatUI")
+class RevenueCatUIPlugin : Plugin(), PaywallResultListener {
+
+    private var savedCall: PluginCall? = null
+    private var customerCenterLauncher: ActivityResultLauncher<Intent>? = null
+
+    override fun load() {
+        super.load()
+
+        // Register an ActivityResultLauncher for the Customer Center flow
+        customerCenterLauncher = activity.registerForActivityResult(
+            ActivityResultContracts.StartActivityForResult()
+        ) {
+            // Regardless of the result we just resolve the call
+            savedCall?.resolve()
+            savedCall = null
+        }
+    }
+
+    @PluginMethod
+    fun presentPaywall(call: PluginCall) {
+        // Check if there's already a presentation in progress
+        if (savedCall != null) {
+            call.reject("A paywall presentation is already in progress")
+            return
+        }
+        
+        savedCall = call
+
+        val offeringIdentifier = call.getString("offeringIdentifier")
+        val displayCloseButton = call.getBoolean("displayCloseButton", false)
+
+        val currentActivity = activity
+        if (currentActivity !is FragmentActivity) {
+            call.reject(
+                "PAYWALLS_MISSING_ACTIVITY",
+                "Paywalls require your activity to subclass FragmentActivity"
+            )
+            return
+        }
+
+        val paywallSource = offeringIdentifier?.let { PaywallSource.OfferingIdentifier(it) }
+            ?: PaywallSource.DefaultOffering
+
+        val options = PresentPaywallOptions(
+            paywallSource = paywallSource,
+            requiredEntitlementIdentifier = null,
+            shouldDisplayDismissButton = displayCloseButton,
+            paywallResultListener = this
+        )
+
+        presentPaywallFromFragment(currentActivity, options)
+        notifyListeners("paywallDisplayed", JSObject())
+    }
+
+    @PluginMethod
+    fun presentPaywallIfNeeded(call: PluginCall) {
```

Comment:
Hmm these 2 methods are pretty similar... I wonder if we can extract some of the logic inside to a shared method, to avoid repeating a lot of the code.